### PR TITLE
feat(Dropdown): builds Dropdown component

### PR DESCRIPTION
**Pull request by hilaryBeck**
[DS: Build Dropdown Functionality](https://app.clubhouse.io/firehydrant/story/14030/ds-build-dropdown-functionality)
[Dropdown styling](https://app.clubhouse.io/firehydrant/story/21858/dropdown-styling)

Adds the Dropdown component based on [React Select](https://react-select.com/home).

### CODE REVIEW

File: lib/components/dropdown/Dropdown.stories.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      },
+    },
+  },
+};
+
+const Template = (args) => {
+  <FormControl>
+    <Dropdown {...args} />
+  </FormControl>
+};
```
Comment (caseymhunt) [inline lib/components/dropdown/Dropdown.stories.js:67]: switch the `{ }` to `( )` and it should fix the render issues.

**Review by caseymhunt — COMMENTED** [review]
From the looks of it, `Dropdown` will not be a conventionally "theme adjustable" component like the components built directly with Chakra. 

For documentation purposes, will there be a new technique to modifying this component (different from the current theme partials) or will this simply be a visually immutable component?

**Comment by hilaryBeck** [comment]
visual immutability i guess is a question for @jaxatto, but yeah, modifying it would be more along the lines of traditional CSS in the `customStyles` than a theme partial

**Comment by caseymhunt** [comment]
> visual immutability i guess is a question for @jaxatto, but yeah, modifying it would be more along the lines of traditional CSS in the `customStyles` than a theme partial

👍🏼  roger. IMO we document this as a component that cannot be modified by conventional methods and **should not** be modified due to its complexity. ¯\_(ツ)_/¯ 

seems fair to me.

**Review by caseymhunt — APPROVED** [review]

**Pull request merged**